.circleci/config.yml

```diff
@@ -270,8 +296,16 @@ workflows:
   version: 2
   release:
     jobs:
-      - build
-      - lints
+      - setup
+      - build:
+          requires:
+            - setup
+      - javascript_build:
+          requires:
+            - setup
```

Comment:
Is `javascript_build` leveraging `setup` currently?

Reply:
Only as a way to ensure dependencies are installed

Reply:
> Only as a way to ensure dependencies are installed

There's been some shuffling of this in recent commits, but since we run `bundle-yarn-install` in `javascript_build`, isn't that happening anyways? It seems like `javascript_build` can work standalone without requiring `setup`.

Reply:
Ah, I misunderstood. `javascript-build` doesn't strictly require `setup` to succeed, no. I made `setup` required for all of the other steps because it does a `bundle-yarn-install`, which will save subsequent steps from potentially downloading a bunch of dependencies.

The scenario in my mind is where the dependencies change and there is no cache yet. If all 7 of the containers across the steps have to download/install without a cache, it'll take more total container time since they'll all do it in parallel. Making all of the steps wait for `setup` means they'll all have a prepared cache they can restore from.

It does potentially increase latency a small amount, but I think the improvement in throughput is worth it to free up a handful of containers sooner.

Reply:
> The scenario in my mind is where the dependencies change and there is no cache yet. If all 7 of the containers across the steps have to download/install without a cache, it'll take more total container time since they'll all do it in parallel. Making all of the steps wait for `setup` means they'll all have a prepared cache they can restore from.

This helped clarify for me and makes a lot of sense. Thanks for explaining 👍